markdown-toc.el

```diff
@@ -222,18 +229,37 @@ If called interactively with prefix arg REPLACE-TOC-P, replaces previous TOC."
   (save-excursion
     (when (markdown-toc--toc-already-present-p)
       ;; when toc already present, remove it
-      (let ((region-start (markdown-toc--toc-start))
-            (region-end   (markdown-toc--toc-end)))
-        (delete-region region-start (1+ region-end))
-        (when replace-toc-p
-          (goto-char region-start))))
+      (markdown-toc--delete-toc))
     (->> (markdown-imenu-create-nested-index)
          markdown-toc--compute-toc-structure
          (funcall markdown-toc-user-toc-structure-manipulation-fn)
          markdown-toc--generate-toc
          insert)))
 
+;;;###autoload
+(defun markdown-toc-generate-or-refresh-toc ()
+  "Generate a TOC for markdown file at current point or refreshes an already generated TOC."
+  (interactive)
+  (markdown-toc-generate-toc t))
+
+;;;###autoload
+(defun markdown-toc-refresh-toc ()
+  "Refreshes an already generated TOC."
+  (interactive)
+  (when (markdown-toc--toc-already-present-p)
+    (markdown-toc-generate-toc t)))
```

Comment:
The idea is you would set `markdown-toc-generate-or-refresh-toc` to a keybinding, and use it to insert a TOC at point if it does not already exist, but if you are in a different part of the file and just want to refresh the TOC you can use the same keybinding without needing to navigate to the TOC header. 

You would use `markdown-toc-refresh-toc` to refresh a TOC only if it was already created, and not insert a new one. This would be usefull in a `before-save` hook on a buffer to make sure a TOC is always up to date.